debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/processor/TransactionCommitConsumer.java

```diff
@@ -144,198 +254,338 @@ private void dispatchChangeEvent(LogMinerEvent event) throws InterruptedExceptio
-                            Object value = ((DmlEvent) event).getDmlEntry().getNewValues()[i];
-                            Object prevValue = ((DmlEvent) prevEvent).getDmlEntry().getNewValues()[i];
-                            if (prevValue == null && value != null) {
-                                LOGGER.trace("\tAdding column index {} to previous SEL_LOB_LOCATOR event", i);
-                                ((DmlEvent) prevEvent).getDmlEntry().getNewValues()[i] = value;
-                            }
-                        }
-                        return true;
-                    }
-                }
+        }
+
+        int length() {
+            return binary ? bytes.length : data.length();
+        }
+
+        int end() {
+            return offset + length();
+        }
+
+        void truncate(int newLength) {
+            assert newLength < length();
+
+            if (binary) {
+                bytes = Arrays.copyOf(bytes, newLength);
+            }
+            else {
+                data = data.substring(0, newLength);
             }
         }
 
-        LOGGER.trace("\tEvent {} is for a different row, merge skipped.", event.getEventType());
-        return false;
-    }
+        void frontTruncate(int newLength) {
+            assert newLength < length();
 
-    private void mergeLobWriteData(LogMinerEvent event) {
-        final Object data;
-        if (this.lastSelectLobLocatorEvent.isBinary()) {
-            // For BLOB we pass the list of chunks as-is to the value converter
-            data = new BlobChunkList(lobWriteData);
+            if (binary) {
+                bytes = Arrays.copyOfRange(bytes, bytes.length - newLength, bytes.length);
+            }
+            else {
+                data = data.substring(data.length() - newLength);
+            }
+            offset += length() - newLength;
         }
-        else {
-            // For CLOB we go ahead and pre-process the list into a single string
-            data = String.join("", lobWriteData);
+
+        void absorb(LobFragment other) {
+            assert other.offset >= offset;
+            assert other.end() <= end();
+            assert other.binary == binary;
+
+            int prefixEnd = other.offset - offset;
+            int suffixStart = other.end() - offset;
+
+            if (binary) {
+                System.arraycopy(other.bytes, 0, bytes, prefixEnd, other.bytes.length);
+            }
+            else {
+                data = data.substring(0, prefixEnd) + other.data + data.substring(suffixStart);
+            }
         }
 
-        final DmlEvent dmlEvent = (DmlEvent) event;
-        final String columnName = lastSelectLobLocatorEvent.getColumnName();
-        final int columnIndex = getSelectLobLocatorColumnIndex();
+        void append(LobFragment other) {
+            assert other.offset >= end();
+            assert other.binary == binary;
```

Comment:
I would prefer if we want to keep these sanity checks, lets turn them into legit DebeziumException raising call sites rather than utilizing `assert` semantics.

Reply:
done